### PR TITLE
Enables safe mode and safe mode polling screens

### PR DESCRIPTION
**Pull request by shawyu-okta**
## Description:

* implement safe mode view
* implement safe mode polling view
* re-enable tests related to safe mode and safe mode polling

## PR Checklist

- [ ] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-546860](https://oktainc.atlassian.net/browse/OKTA-546860)

### CODE REVIEW

File: src/v3/src/types/schema.ts [hunk 1]
```diff
@@ -171,34 +171,58 @@ export interface UISchemaElement {
...
-export interface UISchemaLayout {
+/**
+ * Parent interface for all Layout types with common properties
+ */
+interface Layout {
+  type: string;
+  key?: string;
+  elements: unknown[];
+  options?: Record<string, unknown>;
+}
```
Comment (shawyu-okta) [inline src/v3/src/types/schema.ts:182]: Not sure if this is really helpful so feel free to veto if you think this parent type is more harmful than helpful

File: src/v3/src/types/schema.ts [hunk 2]
```diff
@@ -450,19 +474,6 @@ export interface SuccessCallback extends UISchemaElement {
...
-  options?: {
-    defaultStepIndex: () => number;
-  }
-}
-
-export interface AccordionLayout {
-  type: UISchemaLayoutType.ACCORDION;
-  elements: AccordionPanelElement[];
-}
-
```
Comment (shawyu-okta) [inline src/v3/src/types/schema.ts:465]: just a reordering so the layout types are grouped together

File: src/v3/src/transformer/layout/safeMode/transformSafeModePoll.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  const descriptionElement: DescriptionElement = {
+    type: 'Description',
+    contentType: 'subtitle',
+    options: {
+      content: loc(
+        'poll.form.message',
+        'login',
+        [pollIntervalSec],
+        { $1: { element: 'span' } },
```
Comment (glenfannin-okta) [inline src/v3/src/transformer/layout/safeMode/transformSafeModePoll.ts:55]: 🥳 Love that this is getting usage 🙏

File: src/v3/src/transformer/uischema/updateElementKeys.ts [hunk 4]
```diff
@@ -23,7 +23,18 @@ import { traverseLayout } from '../util';
...
+      if ([
+        UISchemaLayoutType.STEPPER,
+        UISchemaLayoutType.ACCORDION,
+        UISchemaLayoutType.HORIZONTAL,
+        UISchemaLayoutType.VERTICAL,
+      ].includes((element.type) as UISchemaLayoutType)) {
+        return false;
+      }
+      return !!element.type;
+    },
```
Comment (glenfannin-okta) [inline src/v3/src/transformer/uischema/updateElementKeys.ts:37]: nit:
```suggestion
    // never add a synthetic key to layout elements' uischema
    predicate: (element) => ![
        UISchemaLayoutType.STEPPER,
        UISchemaLayoutType.ACCORDION,
        UISchemaLayoutType.HORIZONTAL,
        UISchemaLayoutType.VERTICAL,
      ].includes((element.type) as UISchemaLayoutType),
```

**Review by glenfannin-okta — APPROVED** [review]

File: src/v3/src/transformer/layout/safeMode/transformSafeModePoll.ts (hunk 3 above)
Comment (glenfannin-okta) [reply src/v3/src/transformer/layout/safeMode/transformSafeModePoll.ts:55]: Also minor nit, should this have the `strong` class added to bold the number?

Comment (shawyu-okta) [reply src/v3/src/transformer/layout/safeMode/transformSafeModePoll.ts:55]: I _believe_ we generally don't want to use bold text within strings any more to emphasize text. @alexdahl-okta @narayun-okta can you confirm?

Comment (alexdahl-okta) [reply src/v3/src/transformer/layout/safeMode/transformSafeModePoll.ts:55]: Good question!

If we've landed on a static string (the number doesn't count down), then we probably don't need bold (we don't need to bring attention to a changing value).

I don't think we have a blanket rule against bold, and some draft guidelines from content design have a least one case where bold is recommended. Bold is a helpful tool in the toolbox designers would like to use sometimes.

But I believe there is globalization pressure against bold – it's not native to all writing systems, and makes translation more difficult. So do agree that we can avoid bold unless we have a compelling case for emphasis.

**Pull request merged**